src/rust/engine/async_semaphore/Cargo.toml

```diff
@@ -7,6 +7,8 @@ publish = false
 
 [dependencies]
 parking_lot = "0.11"
+# TODO: See #10291.
```

Comment:
Maybe make the TODO descriptive? What does `Pants hangs in materialize_directories` imply toward future work?

Reply:
All of the usage sites are consistent: there is another instance lower in this file.